docs/README.md

```diff
@@ -0,0 +1,157 @@
+Note that if only one ECU is updated, there is not Secondary ECUs.
+
+#### ecu\_info.yml
+
+ecu_info.yml is the setting file for ECU configuration.
+
+##### File path
+
+/boot/ota/ecu_info.yml
+
+##### Entries
+
+- format_version (string, required)
+
+  This field specifies the ecu_info.yml format.  
+  Currently this field is not used but `1` should be specified for future use.
+
+- ecu_id (string, required)
+
+  This field specifies ECU id and that should be unique in all EUCs.
+
+- ip_addr (string, optional)
+
+  This field specifies this ota client's IP address.  
+  If this field is not specified, "localhost" is used.  
+  NOTE: this IP address is used for the local server bind address.
+
+- secondaries (array, optional)
+
+  This field specifies list of secondary ECUs information.  
+  If this field is not specified, it is treated as if secondary ecu doesn't exist.
+
+  - ecu_id (string, required)
+    ecu id of secondary ECU
+
+  - ip_addr (string, required)
+    IP address of secondary ECU.
+
+- available_ecu_ids (optional)
+  This field specifies a list of ECU id which indicates the target ECU of the ota update.  
+  Only the main ECU should have this information.  
+  If this field doesn't exist, ecu_id value is used as this value.  
+
+  NOTE: The difference between secondaries and available_ecu_ids:  
+  `secondaries` field is the information for listing children's ECUs, not for
+  listing grandchildren. `available_ecu_ids` filed is the information for listing
+  all ECUs. If the child ecu has its own children, and if that child doesn't
+  respond to the parent's request, the grandchildren's information can't be
+  retrieved by the request. That is the reason why available_ecu_ids field is
+  needed.
+
+##### The default setting
+
+If ecu_info.yml doesn't exist, the default setting is used as follows:
+
+- format_version
+  - 1
+
+- ecu_id
+  - "autoware"
+
+#### proxy\_info.yml
+
+proxy_info.yml is the setting file for OTA internal proxy configuration.
+
+##### File path
+
+/boot/ota/proxy_info.yml
+
+##### Entries
+
+- enable_ota_proxy (string, optional)
+
+  This field specifies whether ota proxy is used or not.  
+  If this field is not specified, ota proxy is not used, it means ota client connects to ota server directly.
+
+- gateway (string, optional if enable_ota_proxy is true otherwise not required)
+
+  When the `enable_ota_proxy` field is true, this field specifies whether the ota proxy requests the ota server with HTTPS or HTTP. If true, HTTPS is used otherwise HTTP is used.  
+  If this field is not specified, HTTP is used.
+
+- upper_ota_proxy (string, optional if enable_ota_proxy is true otherwise not required)
+
+  This field specifies the ota proxy address.
+  To specify the ota proxy address, `http://192.168.20.11:8082` notation is used.
+  If this field is not specified, "localhost" is used for accessing ota proxy.
```

Comment:
`upper_ota_proxy` is for configuring ota-proxy in the subECU, only available when `enable_ota_proxy` set to `true`. ota-proxy on sub ECU will send requests to this proxy. 
if `upper_ota_proxy` is not set, the behavior is the ota-proxy will directly requests files from remote ota server without any upper proxy, not using the `localhost`.

So, for main ECU, this option is optional(not required in most case unless the mainECU also doesn't have Internet connection). 
For sub ECUs that don't have Internet access, this option is mandatory, and should be set to the directly connected parent ECU.

Reply:
@Bodong-Yang 
Question about a proxy for a grandchild. Is it possible and how to configure it to support the grandchild?